go.sum

```diff
@@ -1739,7 +1739,6 @@ golang.org/x/lint v0.0.0-20191125180803-fdd1cda4f05f/go.mod h1:5qLYkcX4OjUUV8bRu
 golang.org/x/lint v0.0.0-20200130185559-910be7a94367/go.mod h1:3xt1FjdF8hUf6vQPIChWIBhFzV8gjjsPE/fR3IyQdNY=
 golang.org/x/lint v0.0.0-20200302205851-738671d3881b/go.mod h1:3xt1FjdF8hUf6vQPIChWIBhFzV8gjjsPE/fR3IyQdNY=
 golang.org/x/lint v0.0.0-20201208152925-83fdc39ff7b5/go.mod h1:3xt1FjdF8hUf6vQPIChWIBhFzV8gjjsPE/fR3IyQdNY=
-golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 h1:VLliZ0d+/avPrXXH+OakdXhpJuEoBZuwh1m2j7U6Iug=
 golang.org/x/lint v0.0.0-20210508222113-6edffad5e616/go.mod h1:3xt1FjdF8hUf6vQPIChWIBhFzV8gjjsPE/fR3IyQdNY=
```

Comment:
I was expecting for all the others `golang.org/x/lint` entries to disappear as well as it doesn't seem like we have  an indirect dependency according to go.mod

Reply:
++, to elaborate a bit, that's because there are still indirect dependencies on all these various versions of `golang.org/x/lint`.  You can see them if you run `go mod graph | grep ' golang.org/x/lint'`.  [This article](https://www.ardanlabs.com/blog/2019/12/modules-03-minimal-version-selection.html), particularly Listing 8, explains why older versions of a dependency might still hang around in `go.sum`.